libvast/include/vast/plugin.hpp

```diff
@@ -402,6 +403,47 @@ class rest_endpoint_plugin : public virtual plugin {
     = 0;
 };
 
+// -- loader plugin -----------------------------------------------------
+
+using input_loader = std::function<auto()->generator<chunk_ptr>>;
+using input_parser = std::function<auto(chunk_ptr)->generator<table_slice>>;
```

Comment:
I think the design doc had these as just `loader` and `parser`, respectively. The word `input` seems rather confusing to me here. The signature for the `parser` also changed to `(generator<chunk_ptr>) -> generator<table_slice>`.

Reply:
Unfortunately using `using parser = ... `  creates a conflict with `concept parser = ...` in `parser.hpp:122` which causes a compilation error. 
I can circumvent this error by defining `loader`/`parser` in the `vast::plugins` namespace instead of `vast`.

Reply:
I moved `loader` and `parser` into `vast::plugins` in https://github.com/tenzir/vast/pull/2935/commits/405db6875bd7a26ce4bb440f9010f8335dc84413

Reply:
It doesn't make much sense to me to keep them under `vast::plugins::{loader,parser}`; that is not the appropriate place for them. Can we perhaps either rename the conflicting concept, or move `loader` and `parser` to `vast::loader_plugin::{loader,parser}`?

Please also make sure that at the very least plugin interfaces are commented well.

Reply:
It's easier to move them into `vast::loader_plugin` -> https://github.com/tenzir/vast/pull/2935/commits/ea6253b3f94ce8661ae92c76e3f5fe7eae0e38b7